src/index.js

```diff
@@ -133,3 +126,171 @@ export class WealthscopeSdk {
     }
   }
 }
+
+export class WealthscopeApiClient {
+  /**
+   * Constructor for the Wealthscope Client API.
+   * Stores the authorization token for the current session, and the base URl.
+   * @param {object} opts 
+   */
+  constructor(opts) {
+    this.token = null;
+
+    this.opts = Object.assign(
+        {
+          wealthscopeUrl: 'https://api.staging-bus.wealthscope.ca/v1',
+        },
+        opts
+    );
+  }
+
+  /**
+   * Function that returns the current version from the package.json.
+   * @returns {string}
+   */
+  static version() {
+    return version;
+  }
+
+  /**
+   * This retrieves a valid token and sets this.token to be the valid token.
+   * @param {string} jwtData 
+   * @returns {Promise}
+   */ 
+  login(jwtData) {
+    return fetch(this.opts.wealthscopeUrl + '/auth/authenticate/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({token: jwtData})
+    })
```

Comment:
Should this use `this._getFetchOptions` ?

Reply:
This can't use _getFetchOptions as it hinges on the token being already present to complete.